### ISSUE TITLE
Expose the iteration index in GTScript

### ISSUE THREAD

**Issue by egparedes** [GridTools/gt4py#208]
Expose a new builtin GTScript function exposing the iteration index per axis as a regular `int` value. In the Syntax Workshop, it was decided to use `index(AXES_NAME)` (where AXIS_NAME should be `I`,`J` or `K`)

Example:

```python
with computation(), interval(...):
    field_a = field_b[-2, 0, 0] + field_b[1, 0, 0]
    current_k = index(K)
    if current_k < 3:
         field_a = 1
```

Functionality: the returned value can only be used in the same places as a regular `int` run-time parameter (e.g., not as an offset in a field indexing operation)

**Comment by romanc** [comment]
@FlorianDeconinck / @twicki this issue is (partially) getting addressed in the experimental "THIS_K" functionality, right? I like the idea of extending the concept to all axis.

**Comment by FlorianDeconinck** [comment]
Yes the experimental `THIS_K` is a first pass at this. It's implemented in `debug` and `dace:X` backends for now and live on NASA's experimental branch

**Comment by FlorianDeconinck** [comment]
Poke @oelbert for future discussion on that syntax.
